### PR TITLE
Clean up GuiHandle #459

### PR DESCRIPTION
**Pull request by yamgent**
Part of #459.

Proposed merge commit messages:

```
GuiHandle contains unnecessary parameters in its constructor. It also
contains methods that are more specialized to individual components.

Let's remove these constructor parameters, and move the methods to their
appropriate component handle classes.
```

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@yamgent submitted v1 for review.

* [v1 1/6] [GuiHandle: Remove guiRobot from constructor parameters](https://github.com/se-edu/addressbook-level4/pull/515/commits/94cc72b8c7598851f06d2a1d2c14531f00dbbf18)
* [v1 2/6] [GuiHandle: Remove primaryStage constructor parameter](https://github.com/se-edu/addressbook-level4/pull/515/commits/427271bf0d043960b2ea75ac8108197325a0f0d9)
* [v1 3/6] [GuiHandle: Move pressEnter\(\) to CommandBoxHandle](https://github.com/se-edu/addressbook-level4/pull/515/commits/83bd0e9f9f3c95ef7a985e79552c1332573ebce2)
* [v1 4/6] [GuiHandle: Inline setTextFieldText\(\)](https://github.com/se-edu/addressbook-level4/pull/515/commits/0499f7bc8d9dcf9a9ed51864346aea6a45339c2b)
* [v1 5/6] [GuiHandle: Inline GuiHandle\#getTextFieldText\(String\)](https://github.com/se-edu/addressbook-level4/pull/515/commits/c171784c7b2870b337e3b4177df35900cd3fd185)
* [v1 6/6] [GuiHandle: Inline getTextFromLabel\(String\, Node\)](https://github.com/se-edu/addressbook-level4/pull/515/commits/a1b429fb0b81fcf1b80e4c43f2ddcdad27785a9d)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/48775e45f9e37d5c00b99f7241d2856b3e2eac9e...a1b429fb0b81fcf1b80e4c43f2ddcdad27785a9d))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/515/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by yamgent** [comment]
[v1 5/6] GuiHandle: Inline ~~GuiHandle#~~getTextFieldText(String)

File: src/test/java/guitests/guihandles/CommandBoxHandle.java [hunk 1]
```diff
@@ -60,4 +61,5 @@ public HelpWindowHandle runHelpCommand() {
     public ObservableList<String> getStyleClass() {
         return getNode(COMMAND_INPUT_FIELD_ID).getStyleClass();
     }
+
```
Comment (Zhiyuan-Amos) [inline src/test/java/guitests/guihandles/CommandBoxHandle.java]: unnecessary change

**Comment by yamgent** [comment]
> Maybe change the make the commit message

:O

\*restarts @limmlingg\*

**Comment by CanIHasReview** [comment]
# v2

@yamgent submitted v2 for review.

* [v2 1/6] [GuiHandle: Remove guiRobot from constructor parameters](https://github.com/se-edu/addressbook-level4/pull/515/commits/94cc72b8c7598851f06d2a1d2c14531f00dbbf18)
* [v2 2/6] [GuiHandle: Remove primaryStage constructor parameter](https://github.com/se-edu/addressbook-level4/pull/515/commits/427271bf0d043960b2ea75ac8108197325a0f0d9)
* [v2 3/6] [GuiHandle: Move pressEnter\(\) to CommandBoxHandle](https://github.com/se-edu/addressbook-level4/pull/515/commits/1bcfcced79bd690cea34ea75f75e5d1aaedcb76d)
* [v2 4/6] [Inline GuiHandle\#setTextFieldText\(\)](https://github.com/se-edu/addressbook-level4/pull/515/commits/be7bbe0a2eb46aa37579b8cb61e656995dcc521c)
* [v2 5/6] [Inline GuiHandle\#getTextFieldText\(String\)](https://github.com/se-edu/addressbook-level4/pull/515/commits/61e811e3d5eccb79e616b94ae7014f7fee62e0bd)
* [v2 6/6] [Inline GuiHandle\#getTextFromLabel\(String\, Node\)](https://github.com/se-edu/addressbook-level4/pull/515/commits/0a0f9f6e560fd26946b61e676d362cc874c82cb7)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/48775e45f9e37d5c00b99f7241d2856b3e2eac9e...0a0f9f6e560fd26946b61e676d362cc874c82cb7)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level4/compare/8eb14464b710247873a4c5a1ec0464eacca146e3~1...8eb14464b710247873a4c5a1ec0464eacca146e3))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/515/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by pyokagan** [comment]
The rest of the changes makes sense, thanks.

**Comment by yamgent** [comment]
> I don't quite understand? GuiRobot is not a singleton?

Apologies for the mistake, it was cherry picked from #512 and there was some confusion about which PR got merged.

**Comment by CanIHasReview** [comment]
# v3

@yamgent submitted v3 for review.

* [v3 1/6] [GuiHandle: Remove guiRobot from constructor parameters](https://github.com/se-edu/addressbook-level4/pull/515/commits/4d7c1d1a6ab2c29524729c8b0d34c95c23c067a2)
* [v3 2/6] [GuiHandle: Remove primaryStage constructor parameter](https://github.com/se-edu/addressbook-level4/pull/515/commits/11d62a5712e84194225f8df4f852391f702fbf17)
* [v3 3/6] [GuiHandle: Move pressEnter\(\) to CommandBoxHandle](https://github.com/se-edu/addressbook-level4/pull/515/commits/89e85301e0da4c3403c4b14fd8a134f7acc78cd9)
* [v3 4/6] [Inline GuiHandle\#setTextFieldText\(\)](https://github.com/se-edu/addressbook-level4/pull/515/commits/66be942c9480a4e9225e78c81dd81019d11de445)
* [v3 5/6] [Inline GuiHandle\#getTextFieldText\(String\)](https://github.com/se-edu/addressbook-level4/pull/515/commits/051221111a28e7c842ebdfd941675e90d5c61c83)
* [v3 6/6] [Inline GuiHandle\#getTextFromLabel\(String\, Node\)](https://github.com/se-edu/addressbook-level4/pull/515/commits/f4ce350930625129953cf4d9d5752f3b15f0c611)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/48775e45f9e37d5c00b99f7241d2856b3e2eac9e...f4ce350930625129953cf4d9d5752f3b15f0c611)) ([:chart_with_upwards_trend: Interdiff between v2 and v3](https://github.com/se-edu/addressbook-level4/compare/6a9f4d0d43b0fcf80896a5744f6b41a81addda60~1...6a9f4d0d43b0fcf80896a5744f6b41a81addda60))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/515/3/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by CanIHasReview** [comment]
# v4

@yamgent submitted v4 for review.

* [v4 1/6] [GuiHandle: Remove guiRobot from constructor parameters](https://github.com/se-edu/addressbook-level4/pull/515/commits/94fa0d3d665cb3f3e2a21f7417caa06dfd5572b9)
* [v4 2/6] [GuiHandle: Remove primaryStage constructor parameter](https://github.com/se-edu/addressbook-level4/pull/515/commits/b82b81800f56cebea2ba5e02426bc32d33ea5fe1)
* [v4 3/6] [GuiHandle: Move pressEnter\(\) to CommandBoxHandle](https://github.com/se-edu/addressbook-level4/pull/515/commits/959b779ac47cfe09ee63acde47161dd2320b78e3)
* [v4 4/6] [Inline GuiHandle\#setTextFieldText\(\)](https://github.com/se-edu/addressbook-level4/pull/515/commits/aadf7baefed31fd9848f997511fba53059018605)
* [v4 5/6] [Inline GuiHandle\#getTextFieldText\(String\)](https://github.com/se-edu/addressbook-level4/pull/515/commits/32644baa2b4d3b8bd98806fc3aebf3e32dc36787)
* [v4 6/6] [Inline GuiHandle\#getTextFromLabel\(String\, Node\)](https://github.com/se-edu/addressbook-level4/pull/515/commits/4c7b751aed74aa8876ef3df151bc5b21e5324fd0)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/d4dd4c253b3218faf475d5d3941e5db377e62637...4c7b751aed74aa8876ef3df151bc5b21e5324fd0)) ([:chart_with_upwards_trend: Interdiff between v3 and v4](https://github.com/se-edu/addressbook-level4/compare/15803025f9ac12f8d4e2432e205a9a7933e78bca~1...15803025f9ac12f8d4e2432e205a9a7933e78bca))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/515/4/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by yamgent** [comment]
# v4

Rebased to master.
@pyokagan reminder for review.

File: src/test/java/guitests/guihandles/CommandBoxHandle.java [hunk 2]
```diff
@@ -24,7 +25,10 @@ public void clickOnTextField() {
     }
 
     public void enterCommand(String command) {
-        setTextField(COMMAND_INPUT_FIELD_ID, command);
+        TextField textField = getNode(COMMAND_INPUT_FIELD_ID);
+        guiRobot.clickOn(textField);
+        guiRobot.interact(() -> textField.setText(command));
+        guiRobot.sleep(500);
```
Comment (pyokagan) [inline src/test/java/guitests/guihandles/CommandBoxHandle.java]: The comment ` // so that the texts stays visible on the GUI for a short period` was not transferred here as well fyi, I wonder if that's intentional.

Comment (yamgent) [inline src/test/java/guitests/guihandles/CommandBoxHandle.java]: Is unintentional, thanks for commenting about it. The issue will go away when this PR is rebased after #503 is merged.

**Comment by CanIHasReview** [comment]
# v5

@yamgent submitted v5 for review.

* [v5 1/6] [GuiHandle: Remove guiRobot from constructor parameters](https://github.com/se-edu/addressbook-level4/pull/515/commits/357325788727f5f5e81d10ab07ba50f18532067c)
* [v5 2/6] [GuiHandle: Remove primaryStage constructor parameter](https://github.com/se-edu/addressbook-level4/pull/515/commits/39edc0922122b9d714acc6dbcf46d2d9612715e8)
* [v5 3/6] [GuiHandle: Move pressEnter\(\) to CommandBoxHandle](https://github.com/se-edu/addressbook-level4/pull/515/commits/8eed4fc38775b57ab97bff49709f832519682b54)
* [v5 4/6] [Inline GuiHandle\#setTextFieldText\(\)](https://github.com/se-edu/addressbook-level4/pull/515/commits/99062b465ebd087a0946d396ad7775a714df998d)
* [v5 5/6] [Inline GuiHandle\#getTextFieldText\(String\)](https://github.com/se-edu/addressbook-level4/pull/515/commits/c2319a9a1d377c6d44f4d6b3a90b53fe95228599)
* [v5 6/6] [Inline GuiHandle\#getTextFromLabel\(String\, Node\)](https://github.com/se-edu/addressbook-level4/pull/515/commits/98bab2b3c6a5544e127634a7487bd70b65ef0e17)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/8cfab29d930e3fa60c7411a7eb6992373cff0686...98bab2b3c6a5544e127634a7487bd70b65ef0e17)) ([:chart_with_upwards_trend: Interdiff between v4 and v5](https://github.com/se-edu/addressbook-level4/compare/2e10c7f77e5a287d023dee562f425ad72a037e0e~1...2e10c7f77e5a287d023dee562f425ad72a037e0e))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/515/5/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by yamgent** [comment]
# v5
Rebased.
